tree/30_generic_methods/kernel_module_enabled_at_boot.cf

```diff
@@ -0,0 +1,78 @@
+#######################################################################################
+# Copyright 2019 Normation SAS
+#####################################################################################
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, Version 3.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+#####################################################################################
+
+# @name Kernel module enabled at boot
+# @description Ensure that a given kernel module is enabled on the system
+# @documentation Ensure that a given kernel module is enabled on the system.
+#  Rudder will look for a line matching the module name in a given section in the file:
+#
+#  * `/etc/modules-load.d/enabled_by_rudder.conf` on systemd systems
+#  * `/etc/modprobe.d/enabled_by_rudder.conf` on non-systemd systems
+#
+#  If the module is already enabled by a different option file than used by Rudder, it will add
+#  an entry in the correct file listed above but will let intact the already present one.
+#  The modifications are persistent and made line per line, meaning that Rudder will never remove lines
```

Comment:
"meaning that this Generic Method will never remove lines in the configuration file but only add it if needed" or something like that ?